lib/vector-buffers/src/topology/channel/sender.rs

```diff
@@ -314,14 +314,24 @@ impl<T: Bufferable> Sink<T> for BufferSender<T> {
                 if let Some(instrumentation) = this.instrumentation.as_ref() {
                     instrumentation.try_increment_dropped_event_count(1);
                 }
+                *this.state = SendState::Idle;
```

Comment:
I found it easier to clearer to reset the state in each branch rather than adding another boolean flag to use at the end to determine whether or not to increment the statistics for the buffer.